src/com/dotcms/api/system/user/UserServiceFactory.java

```diff
@@ -259,6 +273,130 @@ public UsersListTemplate(String inode, int permissionType, String filter, int st
 			return results;
 		}
 
+		@Override
+		public User findUserByCompanyAndEmail(final String companyId,
+											  final  String emailAddress) {
+
+			User user = null;
+
+			if (!UtilMethods.isSet(companyId) || !UtilMethods.isSet(emailAddress)) {
+
+				throw new UserException("User can not be null");
+			}
+
+			try {
+
+				if (Logger.isDebugEnabled(this.getClass())) {
+
+					Logger.debug(this, "Finding an user by companyId: " + companyId +
+							", emailAddress: " + emailAddress);
+				}
+
+				user = UserUtil.findByC_EA(companyId, emailAddress);
+			} catch (NoSuchUserException | SystemException e) {
+
+				Logger.error(this, e.getMessage(), e);
+				throw new UserException(e);
+			}
+
+			return user;
+		} // findUserByCompanyAndEmail.
+
+		@Override
+		public User update(final User user) {
+
+			User userUpdated = null;
+
+			if (!UtilMethods.isSet(user)) {
+
+				throw new UserException("User can not be null");
+			}
+
+			try {
+
+				if (Logger.isDebugEnabled(this.getClass())) {
+
+					Logger.debug(this, "Updating the user: " + user.getUserId());
+				}
+
+				userUpdated = UserUtil.update(user);
+			} catch (SystemException e) {
+
+				Logger.error(this, e.getMessage(), e);
+				throw new UserException(e);
+			}
+
+			return userUpdated;
+		} // update.
+
+		@Override
+		public void sendResetPassword(final String companyId,
+									  final String emailAddress,
+									  final Locale locale) throws UserEmailAddressException {
+
+			this.sendResetPassword(companyId, emailAddress, locale, ANGULAR_RESET_PASSWORD_URL_STRATEGY);
+		} // sendResetPassword.
+
+		@Override
+		public void sendResetPassword(final String companyId,
+									  final String emailAddressParam,
+									  final Locale locale,
+									  final UrlStrategy resetPasswordUrlStrategy) throws UserEmailAddressException {
+
+			final User user;
+			final String emailAddress;
+			final String token;
+			final Company company;
+			final String url;
+			final String body;
+			final String subject;
+
+			if (!UtilMethods.isSet(emailAddressParam)) {
+
+				throw new UserEmailAddressException();
```

Comment:
Here will be good add some message, to understand what was the problem. That helps a lot in support to identify what kind of problem we are having